sktime/forecasting/compose/_transform_select_forecaster.py

```diff
@@ -0,0 +1,515 @@
+                    true_for_all = False
+                    break
+
+            # Perform this check for the fallback forecaster too
+            if fallback_forecaster is not None and (
+                tag not in fallback_forecaster.get_tags()
+                or fallback_forecaster.get_tags()[tag] is False
+            ):
+                true_for_all = False
+
+            true_if_all_tags[tag] = true_for_all
+
+        # Extrapolating values for flags that should be True if they are
+        # True for any of the given forecasters!
+        true_if_any_tags = {
+            "requires-fh-in-fit": True,
+            "X-y-must-have-same-index": True,
+        }
+
+        # Create a list of forecasters
+        forecasters = list(self.forecasters.items())
+
+        if self.fallback_forecaster is not None:
+            forecasters.append(("", self.fallback_forecaster))
+
+        # Update the tags by iterating through all tags
+        for tag in true_if_any_tags:
+            self._anytagis_then_set(tag, True, False, forecasters)
+
+        # Update the tags
+        self.set_tags(**true_if_all_tags)
+
+        # Finally, dynamically adding implementation of probabilistic
+        # functions depending on the tags set.
+        if self.get_tags()["capability:pred_int"]:
+            self._predict_interval = _predict_interval
+            self._predict_var = _predict_var
+            self._predict_proba = _predict_proba
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        For the _fit function to work as intended, the transformer
+        must generate and store the extrapolated category in the
+        first column.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : Pd.Series
+            The target time series to which we fit the data.
+
+        fh : ForecastingHorizon | None, optional (default=None)
+            The forecasting horizon with the steps ahead to predict.
+
+        X : Pd.Series | None, optional (default=None)
+            No exogenous variables are used for this.
+
+        Returns
+        -------
+        self : reference to self
+
+        Raises
+        ------
+        ValueError: If the extrapolated category has no provided forecaster
+        and if there is no fallback forecaster provided to the object!
+
+        Example:
+
+        If the passed transformer is an ADICVTransformer(), and the generated
+        series is a lumpy series; however, if there is no key matching "lumpy"
+        in the forecasters parameter, the fallback_forecaster will be used.
+        Additionally, if the fallback_forecaster is None, a ValueError will be thrown.
+        """
+        # passing time series through the provided transformer!
+
+        self.category_ = self.transformer_.fit_transform(X=y).iloc[0, 0]
```

Comment:
I think the args should be `(X=y, y=X)`, in case the transformer takes exogeneous data?

Reply:
added that, hope it does not break anything

Reply:
Thank you so much! I just got off work and was just about to add it in. I'll keep an eye on the tests :)